pkg/core/bundle_list.go

```diff
@@ -57,37 +370,42 @@ func ListBundles(repo string, store storage.Store, opts ...BundleListOption) ([]
 	keyChan := make(chan string)
 	errorChan := make(chan error)
 
-	for i := 0; i < minInt(settings.concurrentBundleList, len(ks)); i++ {
+	// spin workers pool
+	for i := 0; i < minInt(settings.concurrentBundleList, len(keys)); i++ {
 		workers.Add(1)
 		go getBundleAsync(repo, store, keyChan, bundleChan, errorChan, &workers)
 	}
 
-	bds := make([]model.BundleDescriptor, 0, typicalBundlesNum) // preallocate some typical number of bundles, e.g. 1000
+	bds := make(model.BundleDescriptors, 0, settings.bundleBatchSize)
```

Comment:
noticed `len(keys)` instead of `bundleBatchSize` in #267 .. if this is to land as a separate commit, use `len(keys)` everywhere for consistency within `fetchBundleBatch`